test/Elastic.Apm.Tests/SerializationTests.cs

```diff
@@ -378,6 +378,20 @@ public void CustomDeDotting()
 			json.Should().Be("{\"custom\":{\"a_b\":\"customValue1\",\"a_b_c\":\"customValue2\"}}");
 		}
 
+		/// <summary>
+		/// Makes sure that the request bodies that contain json are not changed by serialization.
+		/// It basically tests <see cref="BodyStringSanitizerConverter"/>.
+		/// </summary>
+		[Fact]
+		public void RequestBodyWithJson()
+		{
+			var context = new Context();
+			context.Request = new Request("GET", new Url()) { Body = "{\"foo\": \"bar\"}" };
```

Comment:
With `Body = "{\"foo\": \"k1=v1&k2\"}"` is still fails because `&k2` part is dropped.

Reply:
Oh, my understanding was that something like that we definitely can't and don't want to serialize. 

I thought we only serialize based on the root keys (I'm not sure what the right term here is... so on the thing that is "foo". 

Furthermore this is also json, which I thought we don't serialize. My understanding was that we only serialize formats like `key1=value1&key2=value2`.

Reply:
Oh, I see, on it!

Reply:
I think this made it: https://github.com/elastic/apm-agent-dotnet/pull/617/commits/9be210c15f74f02f38c29a7f29facdea29c89d49 

Thought about check specifically for JSON, but that won't help, because at the end of the day, this could be anything, HTML, XML, or something else.

Now, if as a side-effect we may still sanitize something (I guess that could happen) when it's not in the expected format, then I'd say... then so be it 🤷‍♂ 

After #555 we'll do this better and we can get rid of this while re-parsing thing, which is currently a pain, but I don't see any good way to avoid it with having the body as `object`. (happy to take suggestions though! )

Reply:
I think we should sanitize only known data formats (at this point only form data and we can ask other agents if they sanitize JSON, XML, etc. as a future items) because as it stands now if it's not a valid encoded form data we don't sanitize it that well - for example `Body = "password=my password with equal (=) sign"` will expose `my password with equal (=) sign` as is.